include/ginkgo/core/base/types.hpp

```diff
@@ -371,6 +371,22 @@ GKO_ATTRIBUTES constexpr bool operator!=(precision_reduction x,
     return static_cast<st>(x) != static_cast<st>(y);
 }
 
+/**
+ * Calls a given macro for each executor type for a given kernel.
+ *
+ * The macro should take two parameters:
+ *
+ * -   the first one is replaced with the executor class name
+ * -   the second one with the name of the kernel to be bound
+ *
+ * @param _enable_macro  macro name which will be called
+ * @param _kernel  kernel which will be bound to the operation
+ *
+ * @note  the macro is not called for ReferenceExecutor
+ */
+#define GKO_ENABLE_KERNEL_FOR_ALL_EXECUTORS(_enable_macro, _kernel) \
```

Comment:
Did you replace the `GKO_ENABLE_FOR_ALL_EXECUTORS` with this? Or are both currently in use (just saw that you replaced the previous mentioned one with the marked one in the file `include/ginkgo/core/base/executor.hpp`).

Reply:
`GKO_ENABLE_FOR_ALL_EXECUTORS` is still used in `include/ginkgo/core/base/executor.hpp`. So, both are currently in use and hence there is some code duplication.